File: agents/typescript/0-hello-world/src/agent.ts
import { getDefaultThread, wrapAgent } from "@blaxel/sdk";
import { HumanMessage } from "@langchain/core/messages";
import { CompiledGraph } from "@langchain/langgraph";
import { FastifyRequest } from "fastify";
import { v4 as uuidv4 } from "uuid";

type InputType = {
  inputs: string | null;
  input: string | null;
};

type AgentType = {
  agent: CompiledGraph<any, any, any, any, any, any>;
};

const req = async (request: FastifyRequest, args: AgentType) => {
  const { agent } = args;
  const body = (await request.body) as InputType;
  const thread_id = getDefaultThread(request) || uuidv4();
  const input = body.inputs || body.input || "";
  const responses: any[] = [];

  const stream = await agent.stream(
    { messages: [new HumanMessage(input)] },
    { configurable: { thread_id } }
  );

  for await (const chunk of stream) {
    responses.push(chunk);
  }
  const content = responses[responses.length - 1];
  return content.agent.messages[content.agent.messages.length - 1].content;
};

export const agent = wrapAgent(req, {
  agent: {
    metadata: {
      name: "{{.ProjectName}}",
    },
    spec: {
      model: "{{.Model}}",
      description: "{{.ProjectDescription}}",
    },
  },
});

File: agents/python/0-hello-world/src/agent.py
import uuid
from typing import Union

from blaxel.agents import agent, get_default_thread
from fastapi import Request
from langgraph.graph.graph import CompiledGraph


@agent(
    agent={
        "metadata": {
            "name": "{{.ProjectName}}",
        },
        "spec": {
            "model": "{{.Model}}",
            "prompt": "{{.ProjectPrompt}}",
        },
    }
)
async def main(request: Request, agent: Union[None, CompiledGraph]):
    body = await request.json()
    thread_id = get_default_thread(request) or str(uuid.uuid4())
    agent_config = {"configurable": {"thread_id": thread_id}}
    if body.get("inputs"):
        body["input"] = body["inputs"]

    agent_body = {"messages": [("user", body["input"])]}
    responses = []

    async for chunk in agent.astream(agent_body, config=agent_config):
        responses.append(chunk)
    content = responses[-1]
    return content["agent"]["messages"][-1].content
